sw/device/silicon_creator/manuf/lib/isolated_flash_partition.c

```diff
@@ -0,0 +1,108 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/manuf/lib/isolated_flash_partition.h"
+
+#include "sw/device/lib/dif/dif_flash_ctrl.h"
+#include "sw/device/lib/testing/flash_ctrl_testutils.h"
+
+enum {
+  /**
+   * Isolated partition ID.
+   *
+   * Used for wafer authentication secret.
+   */
+  kFlashInfoPartitionId = 0,
+
+  /**
+   * Isolated parition bank ID.
+   *
+   * Used for wafer authentication secret.
+   */
+  kFlashInfoBankId = 0,
+
+  /**
+   * Isolated partition flash info page ID.
+   *
+   * Used to store the wafer authentication secret
+   */
+  kFlashInfoPageIdWaferAuthSecret = 3,
```

Comment:
In a follow up change, we can move  all the metadata associated with info pages into a separate library (e.g. to capture the mappings used in individualize and personalize).

Reply:
good idea. captured in #18478. marked milestone 2.5.2